Lib/threading.py

```diff
@@ -1504,20 +1504,29 @@ def _shutdown():
 
     global _SHUTTING_DOWN
     _SHUTTING_DOWN = True
-    # Main thread
-    tlock = _main_thread._tstate_lock
-    # The main thread isn't finished yet, so its thread state lock can't have
-    # been released.
-    assert tlock is not None
-    assert tlock.locked()
-    tlock.release()
-    _main_thread._stop()
 
     # Call registered threading atexit functions before threads are joined.
     # Order is reversed, similar to atexit.
     for atexit_call in reversed(_threading_atexits):
         atexit_call()
 
+    # Main thread
+    if _main_thread.ident == get_ident():
+        tlock = _main_thread._tstate_lock
+        # The main thread isn't finished yet, so its thread state lock can't
+        # have been released.
+        assert tlock is not None
+        assert tlock.locked()
+        tlock.release()
+        _main_thread._stop()
```

Comment:
This used to be called before the `atexit_call()` loop, why move it?

Reply:
The doc says that atexit callbacks are called before threads are joined. The "else" block now joins a thread.

Do you prefer to call _main_thread._stop() before calling atexit callbacks? If yes, I can organize the code differently.

Reply:
Thanks.  mostly just making sure it was okay.  it does not appear to matter which order these happen in.  technically i suppose someone could write an atexit callback that wants the main thread's .is_alive() to be True but there is no logical reason to do so.  I'm not worried.